File: leetcode/archived/C++/Best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int low , max = 0;
        if(!prices.size())
            return 0;
        low = prices[0];
        for(int i = 0; i < prices.size(); i++)
        {
            if(prices[i] <= low)
                low = prices[i];
            else
                max = (prices[i] - low) > max ? (prices[i]-low) : max; 
        }
        
        return max;
    }
};

File: leetcode/dp/ Best Time to Buy and Sell Stock.go
package dp

func maxProfit(prices []int) int {
	lowest := prices[0]
	maxPro := 0
	for i := 1; i < len(prices); i++ {
		if prices[i] > lowest {
			maxPro = max(prices[i]-lowest, maxPro)
		} else {
			lowest = prices[i]
		}
	}

	return maxPro
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}
